### PR TITLE
[POAE7-2487] support arrow date32(days) format in cider

### PR DESCRIPTION
**Pull request by kd97**
### What changes were proposed in this pull request?

Use arrow date32(days) in cider and cast into datetime64(seconds) only in time func ​
### Why are the changes needed?

support arrow data format
### Does this PR introduce _any_ user-facing change?

needs to follow arrow data format 

<html>
<body>

Format string | Arrow data type
-- | --
tdD | date32 [days]
ttu | time64 [microseconds] 
tsu | timestamp [microseconds]

</body>
</html>

### How was this patch tested?

UTs
a few tests commented out. TODO:
- Non-group by testcase failed
- cast between date(day32) and string

### CODE REVIEW

File: cider/tests/functionality/DateTypeQueryTest.cpp [hunk 1]
```diff
@@ -72,39 +80,45 @@ class DateRandomAndNullQueryOf2009Test : public CiderTestBase {
...
-  assertQuery("SELECT extract(year from col_b) FROM test", "functions/date/year.json");
-  assertQuery("SELECT extract(quarter from col_b) FROM test",
-              "functions/date/quarter.json");
-  assertQuery("SELECT extract(month from col_b) FROM test", "functions/date/month.json");
-  assertQuery("SELECT extract(day from col_b) FROM test", "functions/date/day.json");
-  assertQuery("SELECT extract(dayofweek from col_b) FROM test",
-              "functions/date/day_of_week.json");
-  assertQuery("SELECT extract(doy from col_b) FROM test",
-              "functions/date/day_of_year.json");
+  assertQueryArrow("SELECT extract(year from col_b) FROM test",
```
Comment (jikunshang) [inline cider/tests/functionality/DateTypeQueryTest.cpp:97]: I think we should keep origin test cases rather than replace it.

**Review by jikunshang — COMMENTED** [review]
need to update document?

File: cider/exec/template/DateTimeIR.cpp [hunk 2]
```diff
@@ -202,21 +202,26 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenDateAdd(
   AUTOMATIC_IR_METADATA(cgen_state_);
   const auto& dateadd_expr_ti = dateadd_expr->get_type_info();
   CHECK(dateadd_expr_ti.get_type() == kTIMESTAMP || dateadd_expr_ti.get_type() == kDATE);
+
   auto datetime = codegen(dateadd_expr->get_datetime_expr(), co, true);
-  auto datetime_fixsize = dynamic_cast<FixedSizeColValues*>(datetime.get());
-  auto datetime_value = datetime_fixsize->getValue();
+  const auto& datetime_ti = dateadd_expr->get_datetime_expr()->get_type_info();
+  auto datetime_nullable = dynamic_cast<FixedSizeColValues*>(datetime.get());
```
Comment (yma11) [inline cider/exec/template/DateTimeIR.cpp:208]: add `CHECK(datetime_nullable )`？

**Comment by yma11** [comment]
@kd97 you need rebase this PR and check the build failure.

File: cider/tests/functionality/DateTypeQueryTest.cpp (hunk 1 above)
Comment (kd97) [reply cider/tests/functionality/DateTypeQueryTest.cpp:97]: doc updated,  and I found it difficult to maintain two sets of codepaths due to the new date type definition, constant/column  value representation, different type storage sizes, so migrated and fixed most  time related test cases in this pr (except string to date func).

**Comment by jikunshang** [comment]
please resolve conflicts. BTW, is this pr still WIP? if not please remove it in title.

**Review by jikunshang — DISMISSED** [review]

**Review by winningsix — APPROVED** [review]

**Comment by kd97** [comment]
> please resolve conflicts. BTW, is this pr still WIP? if not please remove it in title.

resolved conflicts and fixed failed tests.

**Review by jikunshang — APPROVED** [review]

**Review by yma11 — APPROVED** [review]

**Pull request merged**